tests/components/samsungtv/test_media_player.py

```diff
@@ -55,247 +87,322 @@ class UnhandledResponse(Exception):
-        self.device.send_key.assert_called_once_with("KEY_REWIND")
-
-    def test_turn_on(self):
-        """Test turn on."""
-        self.device.send_key = mock.Mock()
-        self.device._mac = None
-        self.device.turn_on()
-        self.device.send_key.assert_called_once_with("KEY_POWERON")
-        self.device._wol.send_magic_packet = mock.Mock()
-        self.device._mac = "fake"
-        self.device.turn_on()
-        self.device._wol.send_magic_packet.assert_called_once_with("fake")
-
 
-@pytest.fixture
-def samsung_mock():
-    """Mock samsungctl."""
-    with patch.dict("sys.modules", {"samsungctl": MagicMock()}):
-        yield
+async def test_volume_up(hass, samsung_mock):
+    """Test for volume_up."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_VOLUME_UP, {ATTR_ENTITY_ID: ENTITY_ID}, True
+    )
+    # assert samsungctl.remote_legacy.RemoteLegacy.control.assert_called_once_with("KEY_VOLUP")
+
+
+async def test_volume_down(hass, samsung_mock):
+    """Test for volume_down."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_VOLUME_DOWN, {ATTR_ENTITY_ID: ENTITY_ID}, True
+    )
+    # assert samsungctl.remote_legacy.RemoteLegacy.control.assert_called_once_with("KEY_VOLDOWN")
+
+
+async def test_mute_volume(hass, samsung_mock):
+    """Test for mute_volume."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    assert await hass.services.async_call(
+        DOMAIN,
+        SERVICE_VOLUME_MUTE,
+        {ATTR_ENTITY_ID: ENTITY_ID, ATTR_MEDIA_VOLUME_MUTED: True},
+        True,
+    )
+    # assert samsungctl.remote_legacy.RemoteLegacy.control.assert_called_once_with("KEY_MUTE")
+
+
+async def test_media_play_pause(hass, samsung_mock):
+    """Test for media_next_track."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_MEDIA_PAUSE, {ATTR_ENTITY_ID: ENTITY_ID}, True
+    )
+    assert hass.states.get(ENTITY_ID).state == STATE_PAUSED
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_MEDIA_PLAY, {ATTR_ENTITY_ID: ENTITY_ID}, True
+    )
+    assert hass.states.get(ENTITY_ID).state == STATE_PLAYING
+
+
+async def test_media_play(hass, samsung_mock):
+    """Test for media_play."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_MEDIA_PLAY, {ATTR_ENTITY_ID: ENTITY_ID}, True
+    )
+    # assert samsungctl.remote_legacy.RemoteLegacy.control.assert_called_once_with("KEY_PLAY")
+
+
+async def test_media_pause(hass, samsung_mock):
+    """Test for media_pause."""
+    await setup_samsungtv(hass, MOCK_CONFIG)
+    assert await hass.services.async_call(
+        DOMAIN, SERVICE_MEDIA_PAUSE, {ATTR_ENTITY_ID: ENTITY_ID}, True
+    )
+    # assert samsungctl.remote_legacy.RemoteLegacy.control.assert_called_once_with("KEY_PAUSE")
```

Comment:
@MartinHjelmare Is this how you want to see it? But don't know how to assert `samsungctl` mock. Can you explain, please?

Reply:
With the change to `samsung_mock` decorator I suggested below, we should be able to assert like this:
```py
mock_control = samsung_mock.remote_legacy.RemoteLegacy.control
assert mock_control.called
assert mock_control.call_args == call("KEY_PAUSE")
```
For `call` see:
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.call

It's safer to assert with mock assertion attributes instead of using mock assertion methods unless we spec the mock. Otherwise a typo in an assertion method will be interpreted as a mock attribute and will not fail.

Reply:
I've the problem that `async_update` is always called additionally to my key calls. How to avoid this? And with other tests I want to call it explicitly, how to do this?

Reply:
Why do you want to avoid having `update` be called?

In entities that have `should_poll` set to `True`, the default, `update` is called during the service call targeting an entity service method. This is to allow feedback from the device at the end of the service call.

Reply:
For polling entities, to have `update` be called in tests we can take control of time and move it forward as needed, by patching time and then fire a time changed event.

See eg:
https://github.com/home-assistant/home-assistant/blob/bfba46d64a6778d6f98bd8b0c44e92f556088b7b/tests/components/directv/test_media_player.py#L541-L591